File: frontend/src/microservices/feedback/objects/EnsureOrderCancellationThreadPlannerRequest.ts
// 本文件定义 feedback 模块的 `EnsureOrderCancellationThreadPlannerRequest`，用于在订单取消场景下确保存在对应的反馈线程。

export type EnsureOrderCancellationThreadPlannerRequest = {
  userId: string
  orderId: string
}

export const ensureOrderCancellationThreadRequestFromJson = (json: string): EnsureOrderCancellationThreadPlannerRequest =>
  JSON.parse(json) as EnsureOrderCancellationThreadPlannerRequest

export const ensureOrderCancellationThreadRequestToJson = (value: EnsureOrderCancellationThreadPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/feedback/objects/EnsureOrderCancellationThreadPlannerRequest.scala
// EnsureOrderCancellationThreadPlannerRequest：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class EnsureOrderCancellationThreadPlannerRequest(userId: String, orderId: String)
object EnsureOrderCancellationThreadPlannerRequest:
  given sourceEncoder: Encoder[EnsureOrderCancellationThreadPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[EnsureOrderCancellationThreadPlannerRequest] = deriveDecoder
